climada/hazard/storm_europe.py

```diff
@@ -199,6 +206,209 @@ def _read_one_nc(self, file_name, centroids):
+        else:
+            time_covered_step = ncdf['time'].diff('time')
+            time_covered_day = time_covered_step.groupby('date').sum()
+            # forecast run should cover at least 18 hours of a day
+            days_to_consider = time_covered_day > np.timedelta64(18,'h') 
+            stacked2 = ncdf.groupby('date'
+                                    ).max().sel(date=days_to_consider
+                                                ).stack(intensity=('y_1',
+                                                                   'x_1'))
+            unique_dates = ncdf['time'].groupby('date').min()
+            considered_dates = unique_dates.sel(date=days_to_consider).values
+
+
+        stacked2 = stacked2.stack(event=('date', 'epsd_1'))
+        stacked2 = stacked2.where(stacked2.VMAX_10M > self.intensity_thres)
+        stacked2 = stacked2.fillna(0)
+
+        # fill in values from netCDF
+        self.intensity = sparse.csr_matrix(stacked2.VMAX_10M.T)
+        self.event_id = np.arange(stacked2.date.size)+1
+
+
+        # fill in default values
+        self.units = 'm/s'
+        self.fraction = self.intensity.copy().tocsr()
+        self.fraction.data.fill(1)
+        self.orig = np.ones_like(self.event_id)*False
+        self.orig[(stacked2.epsd_1 == 0).values] = True
+        self.date = np.repeat(
+            np.array(datetime64_to_ordinal(considered_dates)),
+            ncdf.epsd_1.size
+            )
+        self.event_name = [date_i + '_ens' + str(ens_i)
+                           for date_i, ens_i in zip(date_to_str(self.date),
+                                                    stacked2.epsd_1.values+1)
+                           ]
+        self.frequency = np.divide(
+                np.ones_like(self.event_id),
+                ncdf.epsd_1.size)
+        if not description:
+            description = (model_name +
+                           ' weather forecast windfield ' +
+                           'for run startet at ' +
+                           run_date.strftime('%Y%m%d%H'))
+
+        self.tag = TagHazard(
+                HAZ_TYPE, 'Hazard set not saved, too large to pickle',
+                description=description
+            )
+        # close netcdf file
+        ncdf.close()
+        self.check()
+
+    def read_icon_grib(self, run_date, event_date=None,
+                       model_name='icon-eu-eps', description=None,
+                       grib_dir=None, delete_raw_data=True):
+        """Clear instance and download and read dwd icon weather forecast
+        footprints into it. One event is one full day in UCT. Current setup
+        works for runs starting at 00H and 12H. Otherwise the aggregation is
+        inaccurate, because of the given file structure with 1-hour, 3-hour
+        and 6-hour maxima provided.
+
+        Parameters:
+            run_date (datetime): The starting timepoint of the forecast run
+                of the icon model
```

Comment:
I had a look at the download URLs - it seemed that `run_date` has to correspond to the latest model run available for whichever hour (00, 06, 12, 18) you're requesting (and that the runs become available >3 hr after their timestamp). Is it worth mentioning this in the documentation?

Reply:
Yes, this was fixed in download_icon_grib